pkg/utils/packet.go

```diff
@@ -175,26 +175,27 @@ func AnnounceIPs(ifName string, ipConfigs []*current.IPConfig) error {
 		return fmt.Errorf("invalid Ethernet MAC address: %q", linkObj.Attrs().HardwareAddr)
 	}
 
+	var errResult error
+
 	// For all the IP addresses assigned by IPAM, we will send either a GARP (IPv4) or Unsolicited NA (IPv6).
 	for _, ipc := range ipConfigs {
-		var err error
-		if IsIPv6(ipc.Address.IP) {
+		if IsIPv4(ipc.Address.IP) {
+			err := SendGratuitousArp(ipc.Address.IP, linkObj)
+			if err != nil && errResult == nil {
+				errResult = fmt.Errorf("failed to send GARP message for ip %s on interface %q: %v", ipc.Address.IP.String(), ifName, err)
```

Comment:
In this case, we only sent the last error.

We may want to append all the errors so we can see in debug mode what ips configuration failed.

Reply:
is that better?